dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -94,17 +95,36 @@ def save_partner
       self.regional_partner_id = sanitize_form_data_hash[:regional_partner_id]
     end
 
-    before_save :enroll_user, if: -> {properties_changed?}
+    # override
+    def lock!
+      return if locked?
+      super
+      enroll_user if status == "accepted"
+    end
+
     def enroll_user
       return unless pd_workshop_id
 
-      Pd::Enrollment.find_or_create_by!(
+      enrollment = Pd::Enrollment.where(
         pd_workshop_id: pd_workshop_id,
         email: user.email
-      ) do |enrollment|
-        enrollment.user = user
-        enrollment.school_info = user.school_info
-        enrollment.full_name = user.name
+      ).first_or_initialize
+
+      # If this is a new enrollment, we want to:
+      #   - save it with all required data
+      #   - save a reference to it in properties
+      #   - delete the previous auto-created enrollment if it exists
+      if enrollment.new_record?
+        enrollment.update(
+          user: user,
+          school_info: user.school_info,
```

Comment:
I think we want to create the school info from the application, not the user. Unfortunately we don't have school info directly, so we'll have to construct it from school id or explicit school fields (name, city, state, type) in the case of school id -1 (Other).

In any case, `User.school_info` is less strict than `Pd::Enrollment.school_info` and won't always pass validation in that context.

Reply:
Oh, good catch!

Reply:
Note that enrollments are already being constructed from `user.school_info`, so I'll implement that change as a separate PR